docs/sys-admin/expand-cluster/prepare-and-add-hosts.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Prepare and Add Hosts
+---
+
+# Prepare and Add Hosts
+
+Verify your new host systems are ready for integration into the existing Cloudberry system.
+
+To prepare new host systems for expansion, install the Apache Cloudberry software binaries, exchange the required SSH keys, and run performance tests.
+
+Run performance tests first on the new hosts and then all hosts. Run the tests on all hosts with the system offline so user activity does not distort results.
+
+Generally, you should run performance tests when an administrator modifies host networking or other special conditions in the system. For example, if you will run the expanded system on two network clusters, run tests on each cluster.
+
+:::note
+Preparing host systems for use by a Apache Cloudberry system assumes that the new hosts' operating system has been properly configured to match the existing hosts, described in [Configuring Your Systems](../../cbdb-op-software-hardware.md#supported-os).
+:::
+
+## Add new hosts to the trusted host environment
+
+New hosts must exchange SSH keys with the existing hosts to enable Cloudberry administrative utilities to connect to all segments without a password prompt. Perform the key exchange process twice with the [gpssh-exkeys](../../sys-utilities/gpssh-exkeys.md) utility.
+
+First perform the process as `root`, for administration convenience, and then as the user `gpadmin`, for management utilities. Perform the following tasks in order:
+
+1. [To exchange SSH keys as root](#to-exchange-ssh-keys-as-root)
+2. [To create the gpadmin user](#to-create-the-gpadmin-user)
+3. [To exchange SSH keys as the gpadmin user](#to-exchange-ssh-keys-as-the-gpadmin-user)
+
+:::note
+The Apache Cloudberry segment host naming convention is `sdwN` where `sdw` is a prefix and `N` is an integer ( `sdw1`, `sdw2` and so on). For hosts with multiple interfaces, the convention is to append a dash (`-`) and number to the host name. For example, `sdw1-1` and `sdw1-2` are the two interface names for host `sdw1`.
+:::
+
+### To exchange SSH keys as root
+
+1. Create a host file with the existing host names in your array and a separate host file with the new expansion host names. For existing hosts, you can use the same host file used to set up SSH keys in the system. In the files, list all hosts (coordinator, backup coordinator, and segment hosts) with one name per line and no extra lines or spaces. Exchange SSH keys using the configured host names for a given host if you use a multi-NIC configuration. In this example, `cdw` is configured with a single NIC, and `sdw1`, `sdw2`, and `sdw3` are configured with 4 NICs:
+
+    ```
+    cdw
+    sdw1-1
+    sdw1-2
+    sdw1-3
+    sdw1-4
+    sdw2-1
+    sdw2-2
+    sdw2-3
+    sdw2-4
+    sdw3-1
+    sdw3-2
+    sdw3-3
+    sdw3-4
+    ```
+
+2. Log in as `root` on the coordinator host, and source the `greenplum_path.sh` file from your Cloudberry installation.
```

Comment:
```suggestion
2. Log in as `root` on the coordinator host, and source the `cloudberry-env.sh` file from your Cloudberry installation.
```

Reply:
For the main branch, the file name has been updated. Fox 2.1, we can still use the old one. FYI.